Gems/Pointcloud/Code/Source/Tools/Components/PointcloudEditorComponent.cpp

```diff
@@ -0,0 +1,173 @@
+#include "AzCore/Debug/Trace.h"
+#include <3rd/happly.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzFramework/Entity/EntityContext.h>
+#include <AzFramework/Entity/EntityContextBus.h>
+#include <AzFramework/Physics/Common/PhysicsTypes.h>
+#include <AzFramework/Scene/Scene.h>
+#include <AzFramework/Scene/SceneSystemInterface.h>
+#include <QFileDialog>
+#include <QMessageBox>
+
+#include <AzFramework/Scene/SceneSystemInterface.h>
+
+#include "Clients/PointcloudComponent.h"
+#include <Atom/RPI.Public/Scene.h>
+#include <AzFramework/Scene/SceneSystemInterface.h>
+#include <AzToolsFramework/API/ToolsApplicationAPI.h>
+#include <AzToolsFramework/UI/UICore/WidgetHelpers.h>
+#include <Render/PointcloudFeatureProcessor.h>
+namespace Pointcloud
+{
+
+    void PointcloudEditorComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("TransformService"));
+    }
+
+    void PointcloudEditorComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<PointcloudEditorComponent, AzToolsFramework::Components::EditorComponentBase>()
+                ->Version(2)
+                ->Field("Point Size", &PointcloudEditorComponent::m_pointSize)
+                ->Field("PointcloudAsset", &PointcloudEditorComponent::m_pointcloudAsset)
+                ->Field("Visible", &PointcloudEditorComponent::m_visible)
+                ->Field("NumPoints", &PointcloudEditorComponent::m_numPoints);
+            AZ::EditContext* editContext = serializeContext->GetEditContext();
+            if (editContext)
+            {
+                editContext->Class<PointcloudEditorComponent>("PointcloudEditorComponent", "PointcloudEditorComponent")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "PointcloudEditorComponent")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "RobotecTools")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PointcloudEditorComponent::m_pointSize,
+                        "Point Size",
+                        "Size of the points in the pointcloud")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &PointcloudEditorComponent::OnSetPointSize)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &PointcloudEditorComponent::m_visible, "Visible", "Visibility of the pointcloud")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &PointcloudEditorComponent::OnVisibility)
```

Comment:
shouldn't this be handled by Entity Outliner? By implementing `AzToolsFramework::EditorEntityInfoNotificationBus::OnEntityInfoUpdatedVisibility`?

Reply:
Done, thanks for suggestion!